kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.krad.util.ObjectUtils;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.arizona.kfs.gl.GeneralLedgerConstants;
+import edu.arizona.kfs.gl.batch.service.BudgetAdjustmentCashTransferService;
+import edu.arizona.kfs.gl.businessobject.BudgetAdjustmentTransaction;
+import edu.arizona.kfs.gl.dataaccess.BudgetAdjustmentTransactionDao;
+
+@Transactional
+public class BudgetAdjustmentCashTransferServiceImpl implements BudgetAdjustmentCashTransferService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(BudgetAdjustmentCashTransferServiceImpl.class);
+
+    protected BudgetAdjustmentTransactionDao budgetAdjustmentTransactionDao;
+    protected ObjectCodeService objectCodeService;
+    protected OptionsService optionsService;
+    protected ParameterService parameterService;
+    protected DateTimeService dateTimeService;
+    protected RunDateService runDateService;    
+    protected ReportWriterService reportWriterService;
+    protected UniversityDateService universityDateService;
+    protected String batchFileDirectoryName;
+        
+    /**
+     * Reads an incoming file of general ledger transactions, extracts those transaction that qualify as budget adjustment transactions 
+     * and saves those transactions to holding table GL_BUDGET_ADJUST_TRN_T.
+     */
+    @Override
+    public void extractAndSaveBudgetAdjustmentEntries() {
+    	     
+    	String inputFile = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.BUDGET_ADJUSTMENT_OUTPUT_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+        String errorFile = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.CASH_TRANSFER_TRANSACTIONS_ERROR_OUTPUT_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION; 
+        
+        FileReader INPUT_GLE_FILE = null;
+        PrintStream OUTPUT_ERR_FILE_ps;
+        String GLEN_RECORD;
+        BufferedReader INPUT_GLE_FILE_br;
+        try {
+        	INPUT_GLE_FILE = new FileReader(inputFile);
+        	OUTPUT_ERR_FILE_ps = new PrintStream(errorFile);
+        }
+        catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+         
+        INPUT_GLE_FILE_br = new BufferedReader(INPUT_GLE_FILE);
+        int line = 0;
+        int reportBudgetAdjustDocLoaded = 0;
+        int reportBudgetAdjustDocErrors = 0;
+        int maxSequenceId = 0;
+        String saveDocNumber = new String("");
+        		
+        try {
+        	while ((GLEN_RECORD = INPUT_GLE_FILE_br.readLine()) != null) {
+        		if (!org.apache.commons.lang.StringUtils.isEmpty(GLEN_RECORD) && !org.apache.commons.lang.StringUtils.isBlank(GLEN_RECORD.trim())) {
+                	 line++;
+                	 GLEN_RECORD = org.apache.commons.lang.StringUtils.rightPad(GLEN_RECORD, 183, ' ');
```

Comment:
Why are we right padding with a hard-coded 183 spaces? Maybe add a code comment for further reference.